core/src/main/kotlin/org/evomaster/core/problem/graphql/GraphQLUtils.kt

```
@@ -165,4 +170,273 @@ object GraphQLUtils {
+                                        } else {
+                                            history.remove(ob)
+                                        }
+                                    }
+                                    history.remove(element.tableFieldType)// remove the interface type object to the history:the normal case (not cycle)
+                                } else {
+                                    history.remove(element.tableFieldType)// remove the interface type object to the history:the cycle case
+                                }
+                            }
+                        }
+                    }
+
+                }
+            }
+        }
+        return graph
+    }
+
+    private fun addNode(node: String, graph: MutableMap<String, GraphInfo>): GraphInfo? {
+        return graph.putIfAbsent(node, GraphInfo())
+    }
+
+    private fun addEdge(source: String, destination: String, fieldName: String, graph: MutableMap<String, GraphInfo>): Boolean? {
+        addNode(source, graph)
+        addNode(destination, graph)
+        graph[source]?.edges?.add(destination)
+        return graph[source]?.fields?.add(fieldName)
+    }
+
+    /**
+     * The number of nodes
+     * Note: the count include Query and Mutation
+     */
+    fun getGraphSize(graph: MutableMap<String, GraphInfo>): Int {
+        return graph.size
+    }
+
+    /**
+     * The number of fields that are pointers to other nodes in the graph.
+     * Note: fields in Query and Mutation are considered in this count.
+     */
+    fun getNbrOfEdges(graph: MutableMap<String, GraphInfo>): Int {
+        return graph.values.stream().mapToInt { it.edges.size }.sum()
+    }
+
+    /**
+     * The number of fields in the type Query or mutation
+     */
+    fun getNbrQueriesOrMutations(queriesMutationsEntryPoint: String, graph: MutableMap<String, GraphInfo>): Int? {
+        return graph[queriesMutationsEntryPoint.toLowerCase()]?.fields?.size
+    }
```

better to throw an exception (eg IllegalArgumentException) if `queriesMutationsEntryPoint` is not in the graph